Remove inconsistencies about getting help

kytrinyx opened the pull request:
We references to the unofficial Exercism help channels on
Gitter with references to the new official Exercism forum.

In a couple of places this introduced inconsistencies.

The first is a sample `help.md` file, where instead of
linking to the Gitter channel we referenced a language-
specific sub-channel on the forum. However this was in
a document where we discourage links to Exercism-wide
resources, which will already be included in the docs
seen by the end-user.

The second is the name of a markdown link-reference.
We were linking to the Building Exercism sub-channel,
but called the link-reference 'forum'. Elsewhere,
'forum' links to the top-level forum, and we've called
the reference 'building-exercism' when linking to
this category.

This fixes these inconsistencies.

With thanks to @ee7 for suggesting the improvements in #395

kytrinyx committed:
Remove inconsistencies about getting help

We references to the unofficial Exercism help channels on
Gitter with references to the new official Exercism forum.

In a couple of places this introduced inconsistencies.

The first is a sample `help.md` file, where instead of
linking to the Gitter channel we referenced a language-
specific sub-channel on the forum. However this was in
a document where we discourage links to Exercism-wide
resources, which will already be included in the docs
seen by the end-user.

The second is the name of a markdown link-reference.
We were linking to the Building Exercism sub-channel,
but called the link-reference 'forum'. Elsewhere,
'forum' links to the top-level forum, and we've called
the reference 'building-exercism' when linking to
this category.

This fixes these inconsistencies.

ee7 commented on building/tracks/shared-files.md:
```diff
@@ -56,7 +56,13 @@ This document should **not** link to Exercism-wide (track-agnostic) help resourc
+
+- [Kotlin Documentation](https://kotlinlang.org/docs/reference/)
+- [Kotlin Forums](https://discuss.kotlinlang.org/)
+- [Kotlin Slack Channel](http://kotlinlang.slack.com/): [get invite here](http://slack.kotlinlang.org/)
```
```suggestion
- [Kotlin Slack Channel](https://kotlinlang.slack.com/): [get invite here](https://slack.kotlinlang.org/)
```

kytrinyx replied:
I lifted these examples directly from the Kotlin track. Thanks for the fix.

ee7 commented on building/tracks/shared-files.md:
Non-blocking nit: I don't love "channel", since they're not usually referred to like that. Maybe instead something like:

```suggestion
- [Kotlin Subreddit](https://www.reddit.com/r/kotlin)
```

ee7 commented on building/tooling/analyzers/creating-from-scratch.md:
```diff
@@ -9,7 +9,7 @@ These are the steps to get going:
 
-We have an incredibly friendly and supportive community who will be happy to help you as you work through this! If you get stuck, please start a new topic on [the Exercism forum][forum] or create new issues at [exercism/exercism][exercism-repo] as needed 🙂
+We have an incredibly friendly and supportive community who will be happy to help you as you work through this! If you get stuck, please start a new topic on [the Exercism forum][building-exercism] or create new issues at [exercism/exercism][exercism-repo] as needed 🙂
 
-[forum]: https://forum.exercism.org/c/exercism/building-exercism/125
+[building-exercism]: https://forum.exercism.org/c/exercism/building-exercism/125
```
Could you make the same change in the other two `creating-from-scratch.md` files?

My fault, sorry - I only mentioned the analyzers one in https://github.com/exercism/docs/pull/395#discussion_r1003309144

At the time of this comment, with this PR:

```console
$ git rev-parse --short HEAD
cb1637b
$ git grep --break --heading 'forum.exercism.org'
building/tooling/analyzers/creating-from-scratch.md
14:[building-exercism]: https://forum.exercism.org/c/exercism/building-exercism/125

building/tooling/representers/creating-from-scratch.md
14:[forum]: https://forum.exercism.org/c/exercism/building-exercism/125

building/tooling/test-runners/creating-from-scratch.md
14:[forum]: https://forum.exercism.org/c/exercism/building-exercism/125

building/tracks/new/join-our-community.md
7:[building-exercism]: https://forum.exercism.org/c/exercism/building-exercism/125
8:[forum]: https://forum.exercism.org/

building/tracks/syllabus/README.md
179:[forum-support]: https://forum.exercism.org/c/support/8

mentoring/how_to_give_great_feedback.md
88:If you've got a question about a solution, or just want to bounce ideas with other mentors, jump onto our [forum](https://forum.exercism.org/) and ask away!

using/contact.md
3:The best place to get help with Exercism is in the [forum](https://forum.exercism.org/).
```

ee7 reviewed (approved):
Thanks.

kytrinyx committed:
Use subreddit instead of channel in help doc example

Co-authored-by: ee7 <45465154+ee7@users.noreply.github.com>

kytrinyx committed:
Use https in help doc example

Co-authored-by: ee7 <45465154+ee7@users.noreply.github.com>

kytrinyx committed:
Normalize forum reference

This normalizes the reference to the Building Exercism
subcategory on the forum in markdown reference links.
